src/rules/at-rule-blacklist/index.js

```diff
@@ -0,0 +1,35 @@
+import { isString } from "lodash"
+import { vendor } from "postcss"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "at-rule-blacklist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (name) => `Unexpected function "${name}"`,
+})
+
+export default function (blacklist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+
+    root.walkAtRules(atRule => {
+      if (blacklist.indexOf(vendor.unprefixed(atRule.name).toLowerCase()) === -1) { return }
```

Comment:
There is already one: https://github.com/stylelint/stylelint/pull/1410/files#diff-be9372d80663ea47860b52543fcd97ceR48